optimum/exporters/openvino/stateful.py

```diff
@@ -0,0 +1,146 @@
+#  Copyright 2023 The HuggingFace Team. All rights reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+
+import numpy as np
+from packaging import version
+import openvino as ov
+from openvino.runtime import opset13
+from optimum.intel.utils.import_utils import is_openvino_version
+
+
+def model_has_name(ov_model: ov.Model, name: str):
+    return name in sum([list(t.get_names()) for t in ov_model.inputs + ov_model.outputs], list())
+
+
+def model_has_input(ov_model: ov.Model, name: str):
+    return name in sum([list(t.get_names()) for t in ov_model.inputs], list())
+
+
+def model_has_cache_reorder(ov_model):
+    return model_has_input(ov_model, 'beam_idx')
```

Comment:
The name should be `beam_id` or `beam_ids` if you count batch to stay aligned with `input_ids`

Reply:
Never mind. It's reasonable to state that `idx` is singular form of the word `ids`

Reply:
I wasn't super creative when choosing this name, and just borrowed it from `_reorder_cache` method argument name. So this is aligned with that part of the code that this thing comes from. And this `idx` is not the same as those `id` or `ids` they are indexes for a different thing.